### DISCUSSION TITLE
Em 2020 é ok desenvolver para web usando Windows Insider?

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#1724] [Perguntas e Respostas]
[busca por windows insider no forum](https://github.com/frontendbr/forum/issues?q=windows+insider)

Estou usando o Windows 10 Pro 1909 para desenvolver programas com npm e node, atualmente uso WSL, PowerShell 7, Docker para Windows, VS Code e está tudo ok, não é como no Linux ou no Macos, mas da para fazer muita coisa.

Porém, como sou novo no Windows e não quero ter que configurar os programas novamente (embora seja rápido de se fazer) quero saber se alguém aqui tem usado uma stack parecida porém já com o Windows Insider, **e se tem algum feedback negativo ou positivo geral** da experiencia com essa plataforma beta.

Obrigado! o/

**Comment by felipefialho** [comment]
Talvez o @willgm pode ajudar nisso.

**Comment by valkervieira** [comment]
Imagino que o motivo de querer usar o insider seja para usar o WSL2? Se for, usei por um tempo aqui em casa numa máquina bem fraquinha e fiquei bem impressionado com o quanto melhorou do primeiro para o segundo.

Com o primeiro WSL eu tinha muito problema de lentidão na hora de dar um create-react-app por exemplo, que desapareceram com o WSL2. Não usava o Docker no dia-a-dia do trabalho mas instalei e criei uma meia duzia de containers com MongoDB e postgreSQL e todas funcionavam super bem mas claramente engasgavam na minha RAM (6GB). O nvm funcionava perfeitamente bem e o terminal integrado do VSCode tb, com a extensão que eles te indicam. Ah, os build times também eram muito bons.

O emulador de terminal que eu usei foi o proprio novo terminal do windows e o hyper, ambos funcionavam bem com o zsh que eu sempre instalo no linux, mas eu acabava quase sempre só usando o terminal do vscode mesmo.

Agora, definitivamente não chega perto da experiência do Linux ou do Mac, nesses tempos de quarentena trouxe o Mac Mini do trabalho pra casa e formatei o windows que agora não tem nada pra programar, só uso pro Office e jogos, já deixei dual boot com um Manjaro XFCE pra quando a quarentena acabar, que na mesma máquina performa infinitamente melhor.

Se eu tivesse uma máquina um pouquinho tenho absoluta certeza que teria sido uma experiência bem positiva. Como não é o meu caso acabei desistindo.

Para fins de comparativo minha máquina é um note asus com 6GB de RAM, i5 4210u, SSD 240GB, geforce 840m rodando num monitor ultrawide 2560x1080.

**Comment by leandrocrs** [comment]
### Ainda não, mas falta pouco

Sou desenvolvedor frontend web e entusiasta dessa nova Microsoft. O trabalho do time do WSL é espetacular.

O desempenho é até aceitável, muito atrás do GNU/Linux instalado nativamente na máquina é claro, ou do MacOS, mas muito melhor que a primeira versão do WSL.

Topei até de trabalhar diretamente no Windows quando o WSL2 foi para o Slow Ring, mas me deparei com um problema grave que ainda está em aberto.

O WSL2 possui um memoroy leak ainda não detectado, que o faz consumir 1.5GB sem nenhum container rodando no docker on WSL2, e chegando a consumir até ~8GB com um simples create-react-app em modo de desenvolvimento. Tive que dobrar a quantidade de memória RAM na minha máquina por conta disso, de 8 para 16GB, mas não foi suficiente. Chega em um ponto que a VM simplesmente não tem de onde tirar memória e morre, levando junto todos os processos com ela.

Recomendação: a melhor alternativa ainda é o GNU/Linux puro, para quem precisa de muito desempenho, seguido pelo MacOS para quem trabalha com mobile e frontend, por conta de possuir todos os navegadores, mas o Windows com WSL2 segue como uma alternativa muito boa, assim que corrigirem essa falha de memory leak, que acredito que seja só questão de tempo.

Referências: [WSL 2 consumes massive amounts of RAM and doesn't return it #4166](https://github.com/microsoft/WSL/issues/4166)

**Comment by valkervieira** [comment]
> O WSL2 possui um memoroy leak ainda não detectado, que o faz consumir 1.5GB sem nenhum container rodando no docker on WSL2, e chegando a consumir até ~8GB com um simples create-react-app em modo de desenvolvimento.

@leandrocrs ah isso explica minha experiência então

**Comment by felquis** [comment]
Um problema que apareceu para mim, é que se eu estiver com programas abertos consumindo memória, ao iniciar o Docker para Windows diz "Memória Insuficiente", tenho que parar outros programas e dedicar o PC para o Docker. Uso 8GB de ram, e tenho SSD.

**Comment by leandrocrs** [comment]
> Um problema que apareceu para mim, é que se eu estiver com programas abertos consumindo memória, ao iniciar o Docker para Windows diz "Memória Insuficiente", tenho que parar outros programas e dedicar o PC para o Docker. Uso 8GB de ram, e tenho SSD.

Se possível, faça uma força e dá um upgrade na sua RAM. Pra rodar docker-desktop no MacOS ou Windows, recomendo no mínimo 12GB de RAM.

Pra quem está no GNU/Linux, 8GB são suficientes, já que não há virtualização no processo.

**Comment by popenke** [comment]
Eu vinha usando Ubuntu instalado direto na máquina e passei pro Windows Insider com o WSL2. Roda o Docker no WSL2 e todo o tooling que usava no Linux muito bem, não noto diferença alguma.

Mas isso depois de passar os repositórios todos para o `/home` do WSL, porque no disco C: ele performa mal. Mas de longe, muito MUITO melhor que o WSL1.

Também tenho esse problema do memory leak mas nunca chegou a causar lentidão ou matar algum processo, mas realmente só rodando `wsl --shutdown` para resolver (por enquanto).

No mais a experiência é muito boa mesmo, melhor até que no macOS talvez - all the tooling + all the apps 🚀.

**Comment by leandrocrs** [comment]
> Eu vinha usando Ubuntu instalado direto na máquina e passei pro Windows Insider com o WSL2. Roda o Docker no WSL2 e todo o tooling que usava no Linux muito bem, não noto diferença alguma.
> 
> Mas isso depois de passar os repositórios todos para o `/home` do WSL, porque no disco C: ele performa mal. Mas de longe, muito MUITO melhor que o WSL1.
> 
> Também tenho esse problema do memory leak mas nunca chegou a causar lentidão ou matar algum processo, mas realmente só rodando `wsl --shutdown` para resolver (por enquanto).
> 
> No mais a experiência é muito boa mesmo, melhor até que no macOS talvez - all the tooling + all the apps .

Realmente, tem um grande potencial. Melhor coisa enquanto eu ainda trabalhava no wsl2 era, na hora do almoço, abrir o app do xbox e streamar pro PC, enquanto todo meu trabalho ainda estava lá aberto, só esperando a volta do almoço.

É uma plataforma que estou acompanhando de perto pelo Slow Ring com grande entusiasmo.

**Comment by felquis** [comment]
Tenho sofrido, mas persisto.

É bem legal usar Ubuntu, Windows, MacOS, Android, iPadOS, iOS quando tem que fazer alguma interface da para tirar vantagem de detalhes de cada plataforma embora quase tudo seja semelhante, há diferenças grandes.

E "nenhum" site abre no IE11, meio triste que o Windows 10 venha com o Edge antigo ainda e não o Edge novo.
